From da9e684e852a0da664d6b7d0786a08c586150211 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 01 Oct 2024 08:49:22 +0300
Subject: [PATCH] [FSW-2071] boot_desc: add fields for auto load

Add fields to the image descriptor required for automated images loading
from flash. Also add the flags required to control the automated image
loading sequence.
This patch also parses the added fields from the descriptor.

Change-Id: I3e5083b469801ce3ee61ea8d1895fddf377ca674
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 7a7ad17..04c3718 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -554,6 +554,10 @@
 	desc->desc_ver = desc_ptr->desc_ver;
 	desc->crc32 = desc_ptr->crc32;
 	memcpy(desc->chksum, desc_ptr->chksum, ARRAY_SIZE(desc->chksum));
+	desc->cpu_cfg_base = desc_ptr->cpu_cfg_base;
+	desc->mem_base = ((u64)desc_ptr->mem_base_high << 32) | desc_ptr->mem_base_low;
+	desc->mem_size = desc_ptr->mem_size;
+	desc->flags = desc_ptr->flags.img_flags;
 
 	return 0;
 }
diff --git a/include/boot_desc.h b/include/boot_desc.h
index ae96068..ee189fa 100644
--- a/include/boot_desc.h
+++ b/include/boot_desc.h
@@ -27,6 +27,10 @@
 	u64 loadaddr;
 	u8 chksum[SHA_SUM_LEN];
 	u32 crc32;
+	u32 cpu_cfg_base;
+	u64 mem_base;
+	u32 mem_size;
+	u32 flags;
 };
 
 /*
